rusk-wallet/src/wallet/transaction.rs

```diff
@@ -532,11 +533,23 @@ impl<F: SecureWalletFile + Debug> Wallet<F> {
 
         let stake_pk = BlsPublicKey::from(&stake_sk);
 
-        let reward_amount = state
+        let available_reward = state
             .fetch_stake(&stake_pk)
             .await?
             .map(|s| s.reward)
-            .unwrap_or_default();
+            .ok_or(Error::NoReward)?;
```

Comment:
Now that there is a wallet method we might as well use it:
```suggestion
        let available_reward = self.get_stake_reward(sender_idx).await?;
```